From 4eeabc7836c578c00b3c8e66ae16c7e100e1339d Mon Sep 17 00:00:00 2001
From: William Lai <b04597@freescale.com>
Date: Sun, 19 Sep 2010 16:12:13 +0800
Subject: [PATCH] ENGR00131711-3 ESAI: Output the extal clock directly

Output the extal clock directly for the HCKT, without
any divider. This clock's sample rate is same as ckih
on the board.

Signed-off-by: William Lai <b04597@freescale.com>
---
 sound/soc/imx/imx-esai.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/imx/imx-esai.c b/sound/soc/imx/imx-esai.c
index cd9131e..2154599 100644
--- a/sound/soc/imx/imx-esai.c
+++ b/sound/soc/imx/imx-esai.c
@@ -70,9 +70,9 @@ static int imx_esai_set_dai_sysclk(struct snd_soc_dai *cpu_dai,
 				ecr &= ~(ESAI_ECR_ERI | ESAI_ECR_ERO);
 		} else if (clk_id == ESAI_CLK_EXTAL) {
 				ecr |= ESAI_ECR_ETI;
-				ecr &= ~ESAI_ECR_ETO;
+				ecr |= ESAI_ECR_ETO;
 				ecr |= ESAI_ECR_ERI;
-				ecr &= ~ESAI_ECR_ERO;
+				ecr |= ESAI_ECR_ERO;
 		}
 	}
 
-- 
1.5.4.4

